### PR TITLE
Module 4 - Vagrant Virtual Machine

### PR DESCRIPTION
**Pull request by CharlieCumber**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,12 +18,12 @@ The `.env` file is used by flask to set environment variables when running `flas
 
 ## Running the App
 
-Once the all dependencies have been installed, start the Flask app in development mode within the poetry environment by running:
+To start the vagrant virtual machine by running:
 ```bash
-$ poetry run flask run
+$ vagrant up --provision
```
Comment (hugh-emerson) [inline README.md]: I would say there's no reason to insist the reader uses vagrant. They could still install poetry and python and run `poetry install` / `poetry run flask run` on their actual machine.

It should just be mentioned as an option. Also `--provision` isn't required - that's to force it to rerun any provisioning scripts. In fact, you generally do not want that.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+  SHELL
+
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running TODO app setup script"
+    trigger.run_remote ={privileged: false, inline: "
+      cd /vagrant
+      poetry install
+      poetry run flask run --host 0.0.0.0
```
Comment (hugh-emerson) [inline Vagrantfile]: You can run the app as a background process to allow `vagrant up` to complete.

E.g. `nohup poetry run flask run > app.log 2>&1 &`

**Review by hugh-emerson — APPROVED** [review]
Looks good to me 👍 

How come you are working off a fork of DevOps-Course-Starter-Module-3 instead of continuing with your own to-do app?

**Comment by CharlieCumber** [comment]
I forked from the exercise 3 starter as I struggled to migrate to using poetry from the original fork

**Pull request merged**